internal/apps/config/doctl_test.go

```diff
@@ -0,0 +1,38 @@
+package config
+
+import (
+	"testing"
+	"time"
+
+	"github.com/digitalocean/doctl"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDoctlConfigSource(t *testing.T) {
+	doctlConfig := doctl.NewTestConfig()
+	doctlConfig.Set("", "interactive", true)
+	doctlConfig.Set("last-seen-ago", "bufo", time.Minute)
+
+	// doctl's IsSet implementation pretends namespaces don't exist
+	t.Run("no namespace", func(t *testing.T) {
+		cs := DoctlConfigSource(doctlConfig, "")
+		assert.True(t, cs.IsSet("interactive"))
+		assert.False(t, cs.IsSet("last-seen-ago.bufo")) // namespace not considered by IsSet
+		assert.True(t, cs.IsSet("bufo"))
+
+		assert.Equal(t, true, cs.GetBool("interactive"))
+		assert.Equal(t, time.Minute, cs.GetDuration("last-seen-ago.bufo"))
+		assert.Equal(t, time.Duration(0), cs.GetDuration("bufo")) // does not exist
+	})
+
+	t.Run("yes namespace", func(t *testing.T) {
+		cs := DoctlConfigSource(doctlConfig, "last-seen-ago")
+		assert.True(t, cs.IsSet("interactive"))
+		assert.False(t, cs.IsSet("last-seen-ago.bufo")) // namespace not considered by IsSet
```

Comment:
`IsSet` is still excluded from key mutation:

```go
// DoctlConfigSource converts a doctl.Config into a ConfigSource with an optional default namespace.
func DoctlConfigSource(config doctl.Config, ns string) ConfigSource {
	...

	// doctl expects the namespace to be present on all calls except IsSet.
	return MutatingConfigSource(&doctlConfigSource{config}, mutateKey, []string{"IsSet"}) // <----
}
```

doctl's IsSet implementation doesn't accept namespaces. [The LiveConfig implementation](https://github.com/digitalocean/doctl/blob/5408a2bb2fe7f8a97134074411754b248d967171/doit.go#L268-L288) basically loops through os.Args to find any references to the given key. For example, `doctl apps dev build --build-command` on the CLI maps to `cmdConfig.Doit.GetString("apps.dev.build", "build-command")`. Because IsSet searches the cli args, it wouldn't find a match for `apps.dev.build.build-command` but rather only `build-command`. 

[TestConfig behaves similarly](https://github.com/digitalocean/doctl/blob/5408a2bb2fe7f8a97134074411754b248d967171/doit.go#L456-L466).